dspace-server-webapp/src/test/data/dspaceFolder/config/spring/api/discovery.xml

```diff
@@ -0,0 +1,3722 @@
+<?xml version="1.0" encoding="UTF-8"?>
```

Comment:
I'm a bit worried about duplicate a such important configuration file for testing. For example if we accidentally remove an indexer that is needed by our application we would not note this change in our CI.
Should we refactor such configuration splitting it in more files? one for the mapping, one for the provider, one for each actual configuration? in this way we could add just the "extra configuration" in the test env and override the mapping

Reply:
> Should we refactor such configuration splitting it in more files? one for the mapping, one for the provider, one for each actual configuration

@abollini Good point, but I think this needs some more discussion & feels out of scope for this PR.

For now we can remove the duplicate `discovery.xml` and move the "test configurations" to a separate file?

Reply:
Moved the changes to a `test-discovery.xml`. In this file the `DiscoveryConfigurationService` is overwritten because the additional configuration needs to be added to the map defined in the service. 

As a consequence there are two instances of the service definition but the test one is used correctly in the tests. 

Whenever there are more changes to the mapping, it will still need to be updated in the test file as well, but plugins or the beans representing the configuration  will not need to be copied.